salt/modules/cmdmod.py

```diff
@@ -245,6 +271,8 @@ def _run(cmd,
     '''
     Do the DRY thing and only call subprocess.Popen() once
     '''
+    if 'pillar' in kwargs and not pillar_override:
+        pillar_override = kwargs['pillar']
```

Comment:
It's a little hard to maintain, since we now have to know to look up in this block to check for this condition but other than that, it's clean.